src/test/java/org/kainos/ea/integration/WebServiceIntegrationTest.java

```diff
@@ -98,6 +104,43 @@ void getJobSpecification_shouldReturnErrorStatus404() throws UnsupportedEncoding
                 .request()
                 .get(Response.class);
 
-        Assertions.assertEquals(expectedResponse.getStatus(), response.getStatus());
+        assertEquals(expectedResponse.getStatus(), response.getStatus());
+    }
+
+    @Test
+    void registerUser_shouldReturnID_whenUserIsRegistered() {
+        String generatedString = RandomStringUtils.randomAlphanumeric(10);
+        int expectedResult = HttpStatus.CREATED_201;
+        UserRequest user = new UserRequest(generatedString + "@email.com", "Testpassword!", 1);
+
+        int response = APP.client().target(hostURI + registerEndpoint)
+                .request()
+                .post(Entity.entity(user, MediaType.APPLICATION_JSON_TYPE))
+                .getStatus();
+
+        assertEquals(expectedResult, response);
+    }
+
+    @Test
+    void registerUser_shouldThrow400Error_whenUserCredentialsAreInvalid() {
+        int expectedResult = HttpStatus.BAD_REQUEST_400;
+        UserRequest user = new UserRequest("a", "a", 1);
+
+        int response = APP.client().target(hostURI + registerEndpoint)
+                .request()
+                .post(Entity.entity(user, MediaType.APPLICATION_JSON_TYPE))
+                .getStatus();
+
+        assertEquals(expectedResult, response);
+    }
+
+    @Test
+    void registerUser_shouldThrow404Error_whenDataNotFoundExceptionThrown() {
+        int expectedResult = HttpStatus.NOT_FOUND_404;
+        Response response = APP.client().target( hostURI + registerEndpoint + "/test" )
```

Comment:
this isn't a valid test, that endpoint doesn't exist

Reply:
removed